core/designsystem/src/jvmMain/kotlin/io/github/droidkaigi/confsched/designsystem/component/ProvideSelectionContainerCustomContextMenu.jvm.kt

```diff
@@ -0,0 +1,116 @@
+                state: ContextMenuState,
+                content: @Composable () -> Unit,
+            ) {
+                val webSearchLabel = stringResource(DesignsystemRes.string.web_search)
+
+                // Unless you explicitly add code to close the standard context menu after clicking, the context menu will not close.
+                val closingManager = remember(textManager, state) {
+                    ClosingTextManager(textManager, state)
+                }
+
+                ContextMenuDataProvider(
+                    items = {
+                        val selected = textManager.selectedText.toString().trim()
+                        if (selected.isNotEmpty()) {
+                            listOf(
+                                ContextMenuItem(webSearchLabel) {
+                                    state.status = ContextMenuState.Status.Closed
+
+                                    val encoded = URLEncoder.encode(
+                                        selected,
+                                        StandardCharsets.UTF_8.toString(),
+                                    )
+                                    val url = buildSearchUrl(encoded)
+                                    onWebSearchClick(url)
+                                },
+                            )
+                        } else {
+                            emptyList()
+                        }
+                    },
+                ) {
+                    baseContextMenu.Area(closingManager, state, content)
+                }
+            }
+        },
+        content = content,
+    )
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+private class ClosingTextManager(
+    private val delegate: TextContextMenu.TextManager,
+    private val state: ContextMenuState,
+) : TextContextMenu.TextManager {
+
+    override val selectedText
+        get() = delegate.selectedText
+
+    override val cut: (() -> Unit)?
+        get() = delegate.cut?.let { original ->
+            {
+                state.status = ContextMenuState.Status.Closed
+                original()
+            }
+        }
+
+    override val copy: (() -> Unit)?
+        get() = delegate.copy?.let { original ->
+            {
+                state.status = ContextMenuState.Status.Closed
+                original()
+            }
+        }
+
+    override val paste: (() -> Unit)?
+        get() = delegate.paste?.let { original ->
+            {
+                state.status = ContextMenuState.Status.Closed
+                original()
+            }
+        }
+
+    override val selectAll: (() -> Unit)?
+        get() = delegate.selectAll?.let { original ->
+            {
+                state.status = ContextMenuState.Status.Closed
+                original()
+            }
+        }
+
+    override fun selectWordAtPositionIfNotAlreadySelected(offset: Offset) {
+        delegate.selectWordAtPositionIfNotAlreadySelected(offset)
+    }
+}
```

Comment:
As mentioned in the comments, the standard context menu does not close after clicking, so I implemented it this way.
In the current version of jetbrians.compose, only the copy option seems to be displayed.